### PR TITLE
docs: Add release guide for iceberg-rust

### PR DESCRIPTION
**Pull request by Xuanwo**
Part of https://github.com/apache/iceberg-rust/issues/81

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi @Fokko, if you think this PR is too large, I can split it up. For instance, I could add the scripts in separate PRs.

File: website/src/release.md
Comment (liurenjie1024) [inline website/src/release.md:87]: Do we need this for rust?

File: website/src/release.md
Comment (liurenjie1024) [inline website/src/release.md:89]: How about moving this before `ASF Side` as first step?

File: website/src/release.md
Comment (liurenjie1024) [inline website/src/release.md:172]: Is it possible to put these into `release.sh`

File: website/src/release.md
Comment (liurenjie1024) [inline website/src/release.md:179]: Ditto

File: website/src/release.md
Comment (liurenjie1024) [inline website/src/release.md:93]: I think one missing part is update crates.io?

File: website/src/download.md
Comment (liurenjie1024) [inline website/src/download.md:1]: `iceberg-rust` mainly serves as a library, I think we already have `install.md` to tell user how to use it, so I think maybe we don't need this? If this is telling user to download source from apache website, adding one section in `install.md` would be enough?

**Review by liurenjie1024 — COMMENTED** [review]
Great work, thanks! Left some comments for unknown problems.

File: crates/catalog/hms/DEPENDENCIES.rust.tsv
Comment (Xuanwo) [reply crates/catalog/hms/DEPENDENCIES.rust.tsv:1]: The ASF release requires users to have a complete list of our dependencies to ensure compliance with Apache 2.0 license.

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:87]: Yes. Only source releases in ASF SVN are considered official; all other releases are provided merely for user convenience. ASF adheres to this approach to ensure that ASF releases are always accessible and not restricted by specific vendors.

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:89]: We will vote on the artifacts that have been uploaded.

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:172]: It might be related to the committers' own setup, so I've separated it out.

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:93]: Publishing to crates.io is optional and should be handled by GitHub CI. I'll include a section on this once the crates.io publishing CI setup is complete.

File: website/src/download.md
Comment (Xuanwo) [reply website/src/download.md:1]: > I think maybe we don't need this

ASF release policy requires that users have the ability to download and build from source.

File: website/src/release.md
Comment (liurenjie1024) [reply website/src/release.md:172]: We may not need to put them in `release.sh`, but I would suggest to put them in a script file, so that committer can easily execute them?

File: website/src/download.md
Comment (liurenjie1024) [reply website/src/download.md:1]: > ASF release policy requires that users have the ability to download and build from source.

Then how about adding one section in `install.md` with title `Build from source`? I think they are just one way of install?

File: website/src/release.md
Comment (liurenjie1024) [reply website/src/release.md:93]: Got it, thanks!

File: website/src/reference/setup_gpg.md
Comment (Fokko) [inline website/src/reference/setup_gpg.md:145]: I think we want to `sed 's/opendal/iceberg/g'` this section

File: website/src/release.md
Comment (Fokko) [inline website/src/release.md:211]: I think almost every project uses RC's instead of rounds.
```suggestion
[VOTE] Release Apache Iceberg Rust ${release_version} RC1
```

File: website/src/release.md
Comment (Fokko) [inline website/src/release.md:244]:
```suggestion
To learn more about Apache Iceberg, please see https://rust.iceberg.apache.org/
```

File: website/src/release.md
Comment (Fokko) [inline website/src/release.md:253]: Instead of having this in the mail, what do you think of having this just part of the docs: https://iceberg.apache.org/how-to-release/#validating-a-source-release-candidate

File: website/src/release.md
Comment (Fokko) [inline website/src/release.md:263]: Is it correct that this file is missing?

File: website/src/release.md
Comment (Fokko) [inline website/src/release.md:277]:
```suggestion
[RESULT][VOTE] Release Apache Iceberg Rust ${release_version} RC1
```

File: website/src/release.md
Comment (Fokko) [inline website/src/release.md:358]: I'm a bit of a nitpicker on this one, but I noticed that we don't have a consistent way of capitalizing Apache, Iceberg and Rust.
```suggestion
The Apache Iceberg Rust community is pleased to announce
that Apache Iceberg Rust ${iceberg_version} has been released!
```

Comment (liurenjie1024) [reply website/src/release.md:358]: I'm a little confusing about the phrase `Apache Iceberg Rust community`, maybe we should just say `Apache Iceberg community`?

**Comment by liurenjie1024** [comment]
cc @Xuanwo Any updates?

**Comment by Xuanwo** [comment]
> cc @Xuanwo Any updates?

Sorry for busy in other projects, I will revist this PR later today.

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md]: I'm fine with both of them.

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:253]: I believe it's easier for PMC members to review releases if they have a checklist. But I'm fine if you want to in docs. Do you want to move this part?

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:263]: Added.

Comment (Xuanwo) [reply website/src/release.md:263]: Added.

File: website/src/download.md
Comment (Xuanwo) [reply website/src/download.md:1]: > Then how about adding one section in `install.md` with title `Build from source`? I think they are just one way of install?

It's fine to add a `Build from source` section in install but we still need a seperate download page like https://iceberg.apache.org/releases/

Maybe we can merge intsall to download once we have made one?

File: website/src/release.md
Comment (Xuanwo) [reply website/src/release.md:172]: I feel that writing this script correctly is challenging. How about we leave this change for future PRs?

Comment (liurenjie1024) [reply website/src/release.md:172]: Cool, we can refine it later.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

File: website/src/release.md
Comment (Fokko) [reply website/src/release.md:253]: It is just a personal preference, it is not wrong having it here :)

**Review by Fokko — APPROVED** [review]

**Review by jbonofre — APPROVED** [review]
It looks good to me. Just two minor things:

1. The distribution should limit the file without ASF header. We can add RAT to check that (in `dev` folder for instance). No need to always execute it, but at least when releasing and creating the source distribution. As reminder, ASF requires the src distribution to be available on `dist.apache.org`., this is the minimum artifact that should be distributed (and user can build from there).
2. `NOTICE` file looks incomplete to me. It should list all dependencies included (e.g. distributed) and used (for instance to build/distribute). It's also a good practice to list all a summary of all different licenses used. Regarding `DEPENDENCIES.rust.tsv` files, `NOTICE` looks pretty short. I can help on this :)

Thanks !

**Comment by Xuanwo** [comment]
Hi, @jbonofre, thanks a lot for the reviewing!

> 1. The distribution should limit the file without ASF header. We can add RAT to check that (in `dev` folder for instance). No need to always execute it, but at least when releasing and creating the source distribution. As reminder, ASF requires the src distribution to be available on `dist.apache.org`., this is the minimum artifact that should be distributed (and user can build from there).

We are using [`skywalking-eyes`](https://github.com/apache/skywalking-eyes) to check the ASF headers. And this has been check in all PRs.

> 2. `NOTICE` file looks incomplete to me. It should list all dependencies included (e.g. distributed) and used (for instance to build/distribute). It's also a good practice to list all a summary of all different licenses used. Regarding `DEPENDENCIES.rust.tsv` files, `NOTICE` looks pretty short. I can help on this :)

The `iceberg-rust` library is a pure Rust implementation that doesn't bundle any dependencies in our release. Should we still include dependencies in the NOTICE file?

**Comment by jbonofre** [comment]
@Xuanwo Thanks for your feedback.

I missed `skywalking-eyes`. I will double check in the src distribution.

For the `NOTICE` file, I would at least list the different licenses used (even if not shipped). My understanding (maybe I'm wrong) is that we don't ship any dependencies in the distribution/implementation, but we need (`use`) dependencies.

**Comment by Xuanwo** [comment]
Hi, @Fokko maybe we can merge this PR and move forward?

**Comment by jbonofre** [comment]
I agree, +1 to merge this PR 😄

**Comment by Fokko** [comment]
Fully Agree! Thanks @Xuanwo for working on this! And @jbonofre and @liurenjie1024 for the reviews 🙏

**Pull request merged**
